### ISSUE TITLE
TreeBehavior node level

### ISSUE THREAD

**Issue by jFranky** [cakephp/cakephp#4553]
I think TreeBehavior needs a slight modification. An additional function for calculating node level would be helpfull (for sorting and other stuff, especially in large catalogues). 
I think It could be a separate function for calculating node level using right and left values in behavior or in beforeSave callback of model, by calculating node level on the fly and assigning it to level field of table (maybe "lvl"). I am using similar function myself, but it could work better as a built in functionality...

**Comment by burzum** [comment]
Instead of doing the recursive `getLevel() calls`, wouldn't it be more simple (and faster) to do this?

```
Select count(*) 
FROM table 
WHERE
    lft < 100 AND rght > 100
```

**Comment by burzum** [comment]
The PR just implements a method to calculate the level. I agree with @ADmad that if we implement some code to update the levels on *all* nodes when for example the parent node is changed. This can become a pretty heavy operation on large trees. Remember, nested sets are fast to read but slow to update / insert.

**Comment by markstory** [comment]
Closing as the `getLevel()` method has now been implemented in 3.0
